cdap-runtime-spi/src/main/java/io/cdap/cdap/runtime/spi/profile/WorkerCoreInfo.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.runtime.spi.profile;
+
+public class WorkerCoreInfo {
+
+  public enum PrefixLabel {
```

Comment:
I don't think the building portion should be in SPI. E.g. if later we will want to show range, this would require quite big change in SPI that we want to avoid.

Reply:
i added it here, this was the common repo between app-fabric and data-proc-provisioner. 

I wanted to add this in cdap-proto , but that was not available in dataproc-provisioner

What would be the appropriate location for this?

Reply:
Why do you need app fabric to know about it? It's purely dataproc provisioner business to form the string. The only thing spi needs is the default value

Reply:
App-fabric gets the profile creation request and saves it in structured table. 
And upon profile fetch -- it is directly read from the table. 

We need to add the worker info at the time of creation before saving to table. 
And this all happens in app-fabric

Reply:
added a new branch with string variant. So we can compare 
https://github.com/cdapio/cdap/pull/13873

Reply:
Seems like the right thing is to add a hook in the Provisioner SPI for the Profile http handler to call so that the provisioner implementation (Dataproc in this case) can perform custom profile update logic. The app fabric shouldn’t contains any provisioner specific logic (imagine we have to support this for EMR as well)